design/example.go

```diff
@@ -179,49 +165,115 @@ func (eg *exampleGenerator) generateFormatExample() interface{} {
+func (eg *exampleGenerator) generateValidatedPatternExample() interface{} {
+	for _, v := range eg.a.Validations {
+		if actual, ok := v.(*dslengine.PatternValidationDefinition); ok {
+			example, err := regen.Generate(actual.Pattern)
+			if err != nil {
+				return eg.r.faker.Name()
+			}
+			return example
+		}
 	}
-	return true
+	return nil
 }
 
-func (eg *exampleGenerator) generateValidatedMinValueExample(min float64) interface{} {
-	if eg.a.Type.Kind() == IntegerKind {
-		return int(min) + eg.r.Int()%int(min)
+func (eg *exampleGenerator) hasMinMaxValidation() bool {
+	for _, v := range eg.a.Validations {
+		if _, ok := v.(*dslengine.MinimumValidationDefinition); ok {
+			return true
+		}
+		if _, ok := v.(*dslengine.MaximumValidationDefinition); ok {
+			return true
+		}
 	}
-	return min + eg.r.Float64()*min
+	return false
 }
 
-func (eg *exampleGenerator) checkMaxValueValidation(example interface{}, max float64) bool {
-	if v, ok := example.(int); ok && float64(v) > max {
-		return false
-	} else if v, ok := example.(float64); ok && v > max {
-		return false
+func (eg *exampleGenerator) checkMinMaxValueValidation(example interface{}) bool {
+	for _, v := range eg.a.Validations {
+		switch actual := v.(type) {
+		case *dslengine.MinimumValidationDefinition:
+			if v, ok := example.(int); ok && float64(v) < actual.Min {
+				return false
+			} else if v, ok := example.(float64); ok && v < actual.Min {
+				return false
+			}
+		case *dslengine.MaximumValidationDefinition:
+			if v, ok := example.(int); ok && float64(v) > actual.Max {
+				return false
+			} else if v, ok := example.(float64); ok && v > actual.Max {
+				return false
+			}
+		}
 	}
 	return true
 }
 
-func (eg *exampleGenerator) generateValidatedMaxValueExample(max float64) interface{} {
-	if eg.a.Type.Kind() == IntegerKind {
-		return eg.r.Int() % int(max)
+func (eg *exampleGenerator) generateValidatedMinMaxValueExample() interface{} {
+	min, max := math.Inf(1), math.Inf(-1)
+	for _, v := range eg.a.Validations {
+		switch actual := v.(type) {
+		case *dslengine.MinimumValidationDefinition:
+			min = math.Min(min, float64(actual.Min))
+		case *dslengine.MaximumValidationDefinition:
+			max = math.Max(max, float64(actual.Max))
+		}
+	}
+	if math.IsInf(min, 1) {
+		if eg.a.Type.Kind() == IntegerKind {
+			if max == 0 {
+				return int(max) - eg.r.Int()%3
+			}
+			return eg.r.Int() % int(max)
+		}
+		return eg.r.Float64() * max
+	} else if math.IsInf(max, -1) {
+		if eg.a.Type.Kind() == IntegerKind {
+			if min == 0 {
+				return int(min) + eg.r.Int()%3
+			}
+			return int(min) + eg.r.Int()%int(min)
+		}
+		return min + eg.r.Float64()*min
+	} else if min < max {
+		if eg.a.Type.Kind() == IntegerKind {
+			return int(min) + eg.r.Int()%int(max-min)
+		}
+		return min + eg.r.Float64()*(max-min)
+	} else if min == max {
+		if eg.a.Type.Kind() == IntegerKind {
+			return int(min)
+		}
+		return min
+	} else {
+		panic("Validation: Min > Max")
```

Comment:
Please do not make the code panic when user input is incorrect, panics are only ok in case of bugs. Instead since this is the finalize phase I would just ignore the condition and not generate an example.